src/utils/objectUtils.ts

```diff
@@ -175,3 +175,48 @@ export function deepClone<T extends object | undefined | null>(object: T): T {
   }
   return structuredClone(object)
 }
+
+type TransformToKebabCaseInputType = Record<string, unknown> | null | undefined
+type TransformToKebabCaseReturnType<Input, Output> =
+  Input extends Record<string, unknown> ? Output : Input
+/**
+ * Transforms an object's keys from camelCase or snake_case to kebab-case.
+ * @param object
+ */
+export function transformToKebabCase<
+  Output extends Record<string, unknown>,
+  Input extends TransformToKebabCaseInputType,
+>(object: Input): TransformToKebabCaseReturnType<Input, Output>
```

Comment:
Finally, I decided to get rid of the type inferring part because it was starting to be complex when I tried to handle edge cases (like abbreviations) and difficult to maintain the future -> [check](https://github.com/lokalise/node-core/pull/98/commits/dd2f6c8026d7505aad7c07521356f01145d08254)